addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/internal/protocol/NibeHeatPumpProtocolStates.java

```diff
@@ -26,9 +28,13 @@
         public boolean process(NibeHeatPumpProtocolContext context) {
             if (context.buffer().hasRemaining()) {
                 byte b = context.buffer().get();
-                context.log("Received byte: {}", String.format("%02X", b));
+                if (logger.isTraceEnabled()) {
+                    logger.trace("Received byte: {}", String.format("%02X", b));
+                }
                 if (b == NibeHeatPumpProtocol.FRAME_START_CHAR_FROM_NIBE) {
-                    context.log("Frame start found");
+                    if (logger.isTraceEnabled()) {
```

Comment:
Note that this check is needed if there is no logic inside the statement, I will create a follow up PR for that later